torch/testing/_internal/opinfo_helper.py

```diff
@@ -0,0 +1,110 @@
+from collections import defaultdict
+import warnings
+from functools import partial
+
+import torch
+from torch.testing._internal.common_cuda import (TEST_CUDA)
+from torch.testing._internal.common_methods_invocations import (op_db, _DYNAMIC_DTYPES)
+from torch.testing import (all_types_and_complex_and,
+                           all_types_and_complex,
+                           # all_types_and_half, # not exported by testing at the moment
+                           all_types,
+                           complex_types,
+                           floating_and_complex_types,
+                           # floating_types_and_half, # not exported by testing at the moment
+                           floating_types,
+                           integral_types,
+                           )
+
+DTYPE_DISPATCH = (
+    floating_types,
+    floating_and_complex_types,
+    integral_types,
+    all_types,
+    complex_types,
+    all_types_and_complex,
+)
+
+# Better way to acquire devices?
+DEVICES = ['cpu'] + (['cuda'] if TEST_CUDA else [])
+
+if not TEST_CUDA:
+    warnings.warn("WARNING: CUDA is not available, information pertaining to CUDA could be wrong")
+
+
+class OpInfoHelper:
+    def __init__(self, op_info):
+        self.op = op_info
+        self.sample_inputs_fn = op_info.sample_inputs_func
+
+    def get_supported_dtypes(self):
+        # Return a map of (device, supported_dtypes)
+        device_dtype = defaultdict(list)
+        for device in DEVICES:
+            for dtype in all_types_and_complex_and(torch.bool, torch.bfloat16, torch.half):
+                sample = self.sample_inputs_fn(self.op, device, dtype, False)[0]
```

Comment:
Let's run through all the samples

Reply:
(We should do this in the "(un)supported..." tests, too, but that's a separate issue that we'll address soon.)

Reply:
> but that's a separate issue that we'll address soon.

This might require fixing some operators (look at https://github.com/pytorch/pytorch/issues/58841)

Reply:
It would require fixing their metadata, yes, and ideally we would update the operators, too, to either support the dtype or throw an error saying they didn't support it.